### ISSUE TITLE
It looks like the source code is not aligned with the latest version based on full .NET framework

### ISSUE THREAD

**Issue by risogolo** [dasiths/NEventLite_Core#1]
Can you please keep it updated.
Just trying to understand the implementation, I need to port it to .NET Core, also planning to create provider for elastic search, so far I don't understand if the Eventstore (https://geteventstore.com/) is necessary ? or how can be replaced with custom implementation.

**Comment by dasiths** [comment]
Yes I haven't updated this. Will do it over the weekend.

**Comment by dasiths** [comment]
This repo is the .net core version already. No need to port it. The .net framework version is at https://github.com/dasiths/NEventLite 

I will update it first and then port it to this repo as I find time. See how IEventStorageProvider is implemented if you want to use a different provider. GetEventStore is optional.
public interface IEventStorageProvider
    {
        Task<IEnumerable<IEvent>> GetEventsAsync(Type aggregateType, Guid aggregateId, int start, int count);
        Task<IEvent> GetLastEventAsync(Type aggregateType, Guid aggregateId);
        Task CommitChangesAsync(AggregateRoot aggregate);
    }

	public interface ISnapshotStorageProvider
    {
        int SnapshotFrequency { get; }
        Task<Snapshot.Snapshot> GetSnapshotAsync(Type aggregateType, Guid aggregateId);
        Task SaveSnapshotAsync(Type aggregateType, Snapshot.Snapshot snapshot);
    }

**Comment by risogolo** [comment]
Yes, but Storage providers were not ported, so I'm trying to port them, also I have a need to use Elasticsearch, so If I get "go", I will try to create new provider for Elasticsearch for either EventStorage and SnapshotStorage

**Comment by dasiths** [comment]
Cool. You can certainly give it a go. Fork and then create a pull request once done. Thanks.
